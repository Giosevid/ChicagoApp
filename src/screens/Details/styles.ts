import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#fff',
    width: '100%',
  },
  button: {
    marginTop: 16,
    elevation: 8,
    backgroundColor: '#009688',
    borderRadius: 10,
    paddingVertical: 10,
    paddingHorizontal: 12,
    alignItems: 'center',
    width: '80%',
  },
  favoriteButton: {
    backgroundColor: '#FF00FF',
  },
  textButton: {
    fontSize: 18,
    color: '#fff',
    fontWeight: 'bold',
    alignSelf: 'center',
  },
});

export default styles;
