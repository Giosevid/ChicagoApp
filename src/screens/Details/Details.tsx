import React from 'react';
import {ScrollView, TouchableOpacity, Text} from 'react-native';
import Card from '../../components/Card';
import useArtsQuery from '../../hooks/useArtsQuery';
import styles from './styles';

function Details({route}: any) {
  const {detailScreen} = route.params;
  const {updateIsFavorite} = useArtsQuery();

  const handlerAddToFavorites = () => updateIsFavorite(detailScreen.id);

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <Card {...detailScreen} disabled />
      <TouchableOpacity
        onPress={handlerAddToFavorites}
        style={[
          detailScreen.isFavorite ? styles.favoriteButton : styles.button,
        ]}>
        <Text style={styles.textButton}>
          {detailScreen.isFavorite
            ? 'Quitar de favoritos'
            : 'Agregar a favoritos'}
        </Text>
      </TouchableOpacity>
    </ScrollView>
  );
}

export default Details;
