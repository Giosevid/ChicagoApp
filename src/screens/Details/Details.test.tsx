import {render, screen} from '@testing-library/react-native';
import React from 'react';
import Details, {DetailsProps} from './Details';

describe('Details', () => {
  const defaultProps: DetailsProps = {
    artId: '',
  };

  it('should render', () => {
    const props = {...defaultProps};
    const {queryByText} = render(<Details {...props} />);

    expect(screen.toJSON()).toMatchSnapshot();
    expect(queryByText('Details')).toBeTruthy();
  });
});
