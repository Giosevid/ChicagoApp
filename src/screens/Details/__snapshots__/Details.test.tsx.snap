// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render 1`] = `
<View>
  <Text>
    Details
  </Text>
</View>
`;
